languagetool-language-modules/ga/src/main/java/org/languagetool/rules/ga/DhaNoBeirtRule.java

```diff
@@ -72,7 +72,7 @@ public String getDescription() {
             if ("déag".equalsIgnoreCase(tokens[j].getToken())) {
               markEnd = j;
               replacement = "dháréag";
-              String[] msgTokens = getRestOfNP(tokens, i + 1, j);
+              String[] msgTokens = getRestOfNP(tokens, i + 1, j - 1);
```

Comment:
I have only tested a previous version but this should fix the IndexOutOfBounds exception.
So I don't interfere with your work, tell me when you want me to merge this PR. It is looking better by the day.